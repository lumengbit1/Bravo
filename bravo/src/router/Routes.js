import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import App from './../App';
import Content from './../components/Content';



const Root = () => (
    <div>
       <Switch>
          <Route
             path='/'
             render={props => (
                <App>
                   <Switch>
                      <Route path='/' exact component={Content} />
                      <Route path='/Program' component={Content} />
                      <Route path='/Partners' component={Content} />
                      <Route path='/Locations' component={Content} />
                      <Route path='/AboutMXDC' component={Content} />
                      <Route path='/Contactus' component={Content} />
                      <Route render={() => <Redirect to="/" />} />
                   </Switch>
                </App>
             )}
          />
       </Switch>
    </div>
 );
  

 export default Root;

