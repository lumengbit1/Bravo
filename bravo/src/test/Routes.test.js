import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter,Route  } from 'react-router-dom';
import Enzyme, { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Routes from './../router/Routes'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { Provider} from 'react-redux'

Enzyme.configure({ adapter: new Adapter() });

describe('SampleComponent', () => {
    it('should render', () => {
        let store;
        const middlewares = [thunk];
        const mockStore = configureMockStore(middlewares);
        store=mockStore();
        const component = renderer
        .create(
            <Provider store={store} >
                <MemoryRouter>
                    <Routes />
                </MemoryRouter>
            </Provider>
        )
        .toJSON();
  
      expect(component).toMatchSnapshot();
    });

  });