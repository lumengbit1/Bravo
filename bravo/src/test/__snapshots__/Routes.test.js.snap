// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleComponent should render 1`] = `
<div>
  <div
    className="App"
  >
    <div>
      <div
        className="header"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          <img
            alt="logo"
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "logo.svg",
              }
            }
            style={
              Object {
                "display": "block",
              }
            }
          />
        </a>
        <div
          className="navtext"
        >
          <a>
            <a
              aria-current={null}
              href="/Program"
              onClick={[Function]}
            >
              Program
            </a>
          </a>
          <a>
            <a
              aria-current={null}
              href="/Partners"
              onClick={[Function]}
            >
              Partners
            </a>
          </a>
          <a>
            <a
              aria-current={null}
              href="/Locations"
              onClick={[Function]}
            >
              Locations
            </a>
          </a>
          <a>
            <a
              aria-current={null}
              href="/AboutMXDC"
              onClick={[Function]}
            >
              About MXDC
            </a>
          </a>
          <a>
            <a
              aria-current={null}
              href="/Contactus"
              onClick={[Function]}
            >
              Contact us
            </a>
          </a>
        </div>
        <img
          alt="mobileicon"
          className="mobileicon"
          onClick={[Function]}
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "menu_icon.svg",
            }
          }
          style={
            Object {
              "display": "block",
            }
          }
        />
        <div
          className="mobilemenu"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="maskheader"
          >
            <a
              aria-current="page"
              className="active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              <img
                alt="logo"
                src={
                  Object {
                    "ReactComponent": [Function],
                    "__esModule": true,
                    "default": "logo.svg",
                  }
                }
              />
            </a>
            <img
              alt="close"
              onClick={[Function]}
              src={
                Object {
                  "ReactComponent": [Function],
                  "__esModule": true,
                  "default": "close_icon.svg",
                }
              }
            />
          </div>
          <div
            className="mobilemenutext"
          >
            <a>
              <a
                aria-current={null}
                href="/Program"
                onClick={[Function]}
              >
                Program
              </a>
            </a>
            <a>
              <a
                aria-current={null}
                href="/Partners"
                onClick={[Function]}
              >
                Partners
              </a>
            </a>
            <a>
              <a
                aria-current={null}
                href="/Locations"
                onClick={[Function]}
              >
                Locations
              </a>
            </a>
            <a>
              <a
                aria-current={null}
                href="/AboutMXDC"
                onClick={[Function]}
              >
                About MXDC
              </a>
            </a>
            <a>
              <a
                aria-current={null}
                href="/Contactus"
                onClick={[Function]}
              >
                Contact us
              </a>
            </a>
          </div>
        </div>
      </div>
      <div
        className="headerpicture"
      >
        <h1>
          MXDC Melbourne
        </h1>
        <h3>
          June 20-27
        </h3>
      </div>
    </div>
    <div
      className="content"
    >
      
    </div>
    <div>
      <div
        className="footer"
      >
        <img
          alt="logo"
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "logo.svg",
            }
          }
        />
        <div
          className="footertext"
        >
          <a>
            Privacy
          </a>
          <a>
            Terms and Conditions
          </a>
          <a>
            Frequently Asked Questions
          </a>
          <a>
            Contact Us
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
